controller/scheduler-v2/job.go

```diff
@@ -0,0 +1,179 @@
+package main
+
+import (
+	"errors"
+	"time"
+
+	ct "github.com/flynn/flynn/controller/types"
+	"github.com/flynn/flynn/controller/utils"
+	"github.com/flynn/flynn/host/types"
+)
+
+type JobRequestType string
+type JobState string
+
+const (
+	JobRequestTypeUp   JobRequestType = "up"
+	JobRequestTypeDown JobRequestType = "down"
+	JobStateStarting                  = "starting"
+	JobStateRunning                   = "running"
+	JobStateStopped                   = "stopped"
+	JobStateCrashed                   = "crashed"
+	JobStateRequesting                = "requesting"
+	JobStateNew                       = "new"
+)
+
+type JobRequest struct {
+	*Job
+	RequestType JobRequestType
+	attempts    uint
+}
+
+func NewJobRequest(f *Formation, requestType JobRequestType, typ, hostID, jobID string) *JobRequest {
+	return &JobRequest{
+		Job:         NewJob(f, f.App.ID, f.Release.ID, typ, hostID, jobID),
+		RequestType: requestType,
+	}
+}
+
+func (r *JobRequest) needsVolume() bool {
+	return r.Job.Formation.Release.Processes[r.Type].Data
+}
+
+func (r *JobRequest) Clone() *JobRequest {
+	return &JobRequest{
+		Job:         r.Job.Clone(),
+		RequestType: r.RequestType,
+		attempts:    r.attempts,
+	}
+}
+
+type Job struct {
+	Type      string
+	AppID     string
+	ReleaseID string
+	HostID    string
+	JobID     string
+
+	Formation *Formation
+
+	restarts  uint
+	startedAt time.Time
+	state     JobState
+}
+
+func NewJob(f *Formation, appID, releaseID, typ, hostID, id string) *Job {
+	return &Job{
+		Type:      typ,
+		AppID:     appID,
+		ReleaseID: releaseID,
+		HostID:    hostID,
+		JobID:     id,
+		Formation: f,
+		startedAt: time.Now(),
+		state:     JobStateNew,
+	}
+}
+
+func (j *Job) Clone() *Job {
+	// Shallow copy
+	cloned := *j
+	return &cloned
+}
+
+func (j *Job) IsScheduled() bool {
+	return j.state != JobStateStopped
```

Comment:
This seems to either be named wrong, or the logic is wrong. `IsScheduled` to me sounds like it should indicate that the job is scheduled to start by the scheduler, but is not an active job in the cluster, but this seems to be returning something else? Renaming / commenting would be useful.